pkg/db/matviews.go

```diff
@@ -89,15 +89,10 @@ func syncPostgresMaterializedViews(db *gorm.DB) error {
 		}
 		dropSQL := fmt.Sprintf("DROP MATERIALIZED VIEW IF EXISTS %s", pmv.Name)
 		schema := fmt.Sprintf("CREATE MATERIALIZED VIEW %s AS %s WITH NO DATA", pmv.Name, viewDef)
-		if err := syncSchema(db, hashTypeMatView, pmv.Name, schema, dropSQL); err != nil {
-			return err
-		}
-
-		// Sync index for the materialized view:
 		indexName := fmt.Sprintf("idx_%s", pmv.Name)
 		index := fmt.Sprintf("CREATE UNIQUE INDEX %s ON %s(%s)", indexName, pmv.Name, strings.Join(pmv.IndexColumns, ","))
-		dropSQL = fmt.Sprintf("DROP INDEX IF EXISTS %s", indexName)
-		if err := syncSchema(db, hashTypeMatViewIndex, indexName, index, dropSQL); err != nil {
+		indexDropSQL := fmt.Sprintf("DROP INDEX IF EXISTS %s", indexName)
+		if err := syncSchema(db, hashTypeMatView, pmv.Name, schema, dropSQL, &index, &indexDropSQL); err != nil {
```

Comment:
Great find. 

This looks like we lose the ability to adjust an index in absence of a mat view change. That might be ok, but would it be simple to just return a bool for changed from the matview syncSchema call, and if true, assume to recreate ALL indexes?

Reply:
That solution has the same problem, if you add just an index it won't get updated if the underlying matview doesn't change.

Reply:
I thought I was hashing indicies, so if they changed in isolation, we'd update them.